hedgehog/src/Hedgehog/Internal/Property.hs

```diff
@@ -550,24 +574,85 @@ assert b = do
   else
     withFrozenCallStack failure
 
+-- | Fails the test if the relation evaluates to 'False' when
+--   applied to its arguments.
+--
+relationWith :: (
+    MonadTest m
+  , Show a
+  , Show b
+  , HasCallStack
+  )
+  => Maybe String
+  -> Maybe String
+  -> (a -> b -> Bool)
+  -> a
+  -> b
+  -> m ()
+relationWith mheader mnegated relation_ x y = do
+  ok <- withFrozenCallStack $ eval (x `relation_` y)
+  if ok then
+    success
+  else
+    withFrozenCallStack $ failDiff mheader mnegated x y
+
+-- | Fails the test if the relation evaluates to 'False' when
+--   applied to its arguments.
+--
+--   /Use 'assertRelationWith' to customize the error message.'
+--
+relation :: (
+    MonadTest m
+  , Show a
+  , Show b
+  , HasCallStack
+  )
+  => (a -> b -> Bool)
+  -> a
+  -> b
+  -> m ()
+relation relation_ x y =
+  withFrozenCallStack $ relationWith Nothing Nothing relation_ x y
+
 infix 4 ===
 
 -- | Fails the test if the two arguments provided are not equal.
 --
+--   /Synonym for '@=='./
+--
 (===) :: (MonadTest m, Eq a, Show a, HasCallStack) => a -> a -> m ()
-(===) x y = do
-  ok <- withFrozenCallStack $ eval (x == y)
-  if ok then
-    success
-  else
-    withFrozenCallStack $ failDiff x y
+(===) x y =
+  withFrozenCallStack $ x @== y
 
 infix 4 /==
 
 -- | Fails the test if the two arguments provided are equal.
 --
+--   /Synonym for '@/='./
+--
 (/==) :: (MonadTest m, Eq a, Show a, HasCallStack) => a -> a -> m ()
-(/==) x y = do
+(/==) x y =
+  withFrozenCallStack $ x @== y
+
+infix 4 @==
+
+-- | Fails the test if the two arguments provided are equal.
```

Comment:
"not equal"? (And I think the `@/=` comment is the wrong way around too)

Reply:
yeah, you're right oops

i think i might remove the operators anyway after chatting to a few people about this PR